src/libasr/asr_utils.h

```diff
@@ -2773,23 +2774,40 @@ inline int extract_kind(ASR::expr_t* kind_expr, const Location& loc) {
                 LCOMPILERS_ASSERT( ASR::is_a<ASR::IntegerConstant_t>(*kind_isf->m_value) );
                 ASR::IntegerConstant_t* kind_ic =
                     ASR::down_cast<ASR::IntegerConstant_t>(kind_isf->m_value);
-                a_kind = kind_ic->m_n;
+                return kind_ic->m_n;
             } else {
                 throw SemanticError("Only Integer literals or expressions which "
                     "reduce to constant Integer are accepted as kind parameters.",
                     loc);
             }
-            break;
         }
-        default: {
+        // allow integer binary operator kinds (e.g. '1 + 7')
+        case ASR::exprType::IntegerBinOp:
+        // allow integer kinds (e.g. 4, 8 etc.)
+        case ASR::exprType::IntegerConstant: {
+            // we still need to ensure that values are constant
+            // e.g. 'integer :: a = 4; real(1*a) :: x' is an invalid kind,
+            // as 'a' // isn't a constant.
+            // ToDo: we should raise a better error, by "locating" just
+            // 'a' as well, instead of the whole '1*a'
+            int a_kind {};
```

Comment:
Initialize to `-1` instead?
```suggestion
            int a_kind = -1;
```

Reply:
I've changed its initialisation to -1.